conn.go

```diff
@@ -378,10 +379,19 @@ func (s *startupCoordinator) setupConn(ctx context.Context) error {
 	}
 	defer cancel()
 
+	// Only for proto v5+.
+	// Indicates if STARTUP has been completed.
+	// github.com/apache/cassandra/blob/trunk/doc/native_protocol_v5.spec
+	// 2.3.1 Initial Handshake
+	// 	In order to support both v5 and earlier formats, the v5 framing format is not
+	//  applied to message exchanges before an initial handshake is completed.
+	startupCompleted := &atomic.Bool{}
```

Comment:
This will cause the minimum Go version to be 1.19 (instead of the current 1.17 according to my experiments). I raised this on the ASF slack channel to see if people think it's ok otherwise we might have to use `atomic.Value` instead...

Reply:
Oops, I missed it. It's kinda a shame that we can't use more distinguishing atomic types instead of generic `Value`... However, the code base already has some similar things, so changing it is not a very big problem.

Reply:
Yeah I'd rather use `atomic.Bool` and `1.19` is 3 versions behind what we currently officially support anyway so personally I don't think it is a big deal to bump the version on `go.mod` to `1.19` but let's see if others chime in

Reply:
Then I'll leave it by the time when we receive some feedback on it

Reply:
I'm going to keep this one open until we have a second reviewer to look at this.